crates/iceberg-catalog/src/implementations/postgres/tabular/table.rs

```diff
@@ -222,41 +145,45 @@ pub(crate) async fn create_table(
             .build()
     })?;
 
+    let tabular_id = create_tabular(
+        CreateTabular {
+            id: table_id.into_uuid(),
+            name,
+            namespace_id: namespace_id.into_uuid(),
+            typ: TabularType::Table,
+            metadata_location: metadata_location.map(std::string::String::as_str),
+            location: location.as_str(),
+        },
+        &mut *transaction,
+    )
+    .await?;
+
     // ToDo: Should we keep the old table_id?
```

Comment:
No, it is only correct this way.
Lets assume two processes try in parallel to create a table.
1. A creates table staged
2. B creates table staged (overwrites)
3. A commits table - A sends a AssertTableUuid as part of the request to determine if it's still the same table. It should not match.

So a new Uuid communicates that someone else overwrote their staged table and the commit cannot work anymore.

Reply:
This ToDo is copied from current main, I'll remove it then?

Reply:
removed ToDo